src/Adapter/Order/CommandHandler/IssuePartialRefundHandler.php

```diff
@@ -74,14 +74,16 @@ public function handle(IssuePartialRefundCommand $command)
         $amount = 0;
         $orderDetailList = [];
         $fullQuantityList = [];
+        $taxCalculator = $this->getTaxCaculator($order->carrier_tax_rate);
 
         foreach ($refunds as $orderDetailId => $refund) {
-            $quantity = $refund['quantity'];
-
-            if (!$quantity) {
-                continue;
+            // this refund has an amount but no quantity, this should not happen
+            if (empty($refund['quantity'])) {
+                throw new OrderException('Please enter a quantity to proceed with your refund.');
```

Comment:
Does the form type allows this? It seems the form type has required false which is why the value can be empty, you can add some constraints in the FormType to force values to be filled
The advantage is that you can then take advantage of Symfony form error display but then you need to modify your controller so that it renders the page with the form in case of validation error (for now it only use flash messages and redirects)

If you don't want to use the form validation but prefer to validate data in the handler through an exception then you need to use a specific exception so that you can manage its translation in the controller for your flash message The way it is now the error message will always be in english since you display the error message as is

You would have to use `$this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));` in the controller, and add the exception treatment in `getErrorMessages`

Reply:
So I noticed my mistake and fixed it while you wrote this message.

About the form type, the field is required => false because it can indeed be empty, the thing is: we can't have empty `quantity` AND a not empty `amount` field for a same product.

What I did so far in the handler is to throw an exception with an error message, and the error is caught in the controller, the message is displayed via the `addFlash` method.